### DISCUSSION TITLE
how to style in dark mode using tailwind?

### DISCUSSION THREAD

**Discussion by glfeng318** [jacobobryant/biff#194] [General]
descripts here: [**toggling-dark-mode-manually in tailwind**](https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually)
it seems biff/base-html dont have a way to set class on html tag, or i miss something?

**Comment by jacobobryant** [comment]
Yeah, `biff/base-html` doesn't have any options to set class on `<html>`. In general, if you ever need something `base-html` doesn't provide you can just copy [the source](https://github.com/jacobobryant/biff/blob/146f2b1c8e0563bd288795851beae8b985cacac7/src/com/biffweb/impl/rum.clj#L29) into your project and make the changes there.

However in this case, you don't actually need to set `<html class="dark"` on the initial server-side render--instead you probably want a bit of javascript that checks client-side if dark mode should be enabled and then adds the class. I've written a how-to for this actually: https://gist.github.com/jacobobryant/14834ada08b00242b28c2b36f4c0225d (from the [content library](https://biffweb.com/docs/library/))
